### ISSUE TITLE
Tab to Space issue when set to 3 spaces

### ISSUE THREAD

**Issue by Relativity70** [notepad-plus-plus/notepad-plus-plus#12528]
### Description of the Issue

Tab size of 3 doesn't work when using Blank Operations / Tab to Space.  Tab Size 2 results in 2 spaces, Tab Size 4 results in 4 spaces, but Tab Size 3 results in 1 space.

### Steps to Reproduce the Issue

1. Settings / Preferences / Language / Tab size => set to 3
2. Copy / Paste cells from Excel into Notepad++
3. Edit / Blank Operations / TAB to Space

### Expected Behavior

TAB would be replaced by three spaces
### Actual Behavior

TAB was replaced by a single space
### Debug Information

Notepad++ v8.4.7   (64-bit)
Build time : Nov  1 2022 - 23:31:11
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Pro for Workstations (64-bit) 
OS Version : 21H2
OS Build : 19044.2130
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)

![tab size 3](https://user-images.githubusercontent.com/118376634/202231301-fa15616b-8232-4e14-879c-f69f20d64d8a.JPG)
![single space](https://user-images.githubusercontent.com/118376634/202231303-431cd044-c7b8-4f80-8aaa-fd404c74fff0.JPG)

**Comment by alankilborn** [comment]
The number of spaces inserted when you press the Tab key is governed by what column you are in when you press it.  Take that column number, subtract 1, and divide by 3 (3 is for your stated configuration).  If the remainder from that math operation is 0, the Tab press will insert 3 spaces.  If the remainder is 2, the Tab press will insert 1 space.  And finally, if the remainder is 1, the Tab press will insert 2 space characters.

In other words, pressing Tab will insert the correct amount of spaces to get you to the next "tab stop" column.  For 3-space tabs, those column numbers are 4, 7, 10, 13, etc.

The _Tab to Space_ operation does a similar thing, but it is easier to think about it first as if you are manually using the Tab key.

Please CLOSE this issue, as Notepad++ is operating as designed in this regard.

**Comment by alankilborn** [comment]
>Tab to Space operation and its use of Tab size is intentionally unintuitive

It is NOT unintuitive.
In fact, the root of how it works is older than me for sure, and likely older than you.
Do some research (on "tab stop") if you're curious; start [HERE](https://en.wikipedia.org/wiki/Tab_stop).

**Comment by Relativity70** [comment]
I appreciate the background information and clearly I misunderstood the intent of an operation called "TAB to Space".  I expected it would replace each tab with a set number of spaces and in practice it may or may not actually do that.  In my case a setting of 1 tab = 2 spaces resulted in every tab being replaced by 2 spaces.  A setting of 1 tab = 4 spaces similarly resulted in every tab being replaced by 4 spaces.  So far so good, until I set it to 1 tab = 3 spaces then most, but not all, were replaced by only 1 space.  In my brief experience this was not an intuitive result.  From what you stated the predictable results of 2 and 4 were likely coincidental to the data I was using and which column numbers the tabs landed on.  A setting of 3 wasn't so lucky.

As you stated Search and Replace works much better for this operation and I'll stick to that going forward. 

Tab stops work as expected, my report was exclusive to the TAB to Space operation and the Tab size setting.  Good day.

**Comment by alankilborn** [comment]
Well, if we're going to be fixated on the name of the command, _TAB to Space_ perhaps should really convert one tab character to exactly **one** space character. :-)

In practice, this command really isn't used for tab characters embedded between other text, it is used for converting leading tab characters to the correct number of spaces.  With leading tab characters, you'd always obtain the "correct" number of spaces in the replacement.

So that opens to debate why there isn't a second option for _TAB to Space_ for "leading", like there is for the other way around, reference:

![image](https://user-images.githubusercontent.com/77065706/202278908-32cd8bfd-52b0-488f-baae-b7243d2fa56d.png)

The answer to that is (probably) that those people wanting to eradicate tab characters have a historical passion about it, and would NEVER want to leave tab characters embedded between non-leading text, after they've eliminated leading tabs, so the single command exterminates all tab characters at once.

A lot of this is very historical, and Notepad++ is first and foremost a "coder's" editor, and a lot of this behavior has its age-old roots in coding.

I hope this has satisfied you, but I have my doubts.  :-(

For your case, the bottom line is that you just "used the wrong command".  Unknowingly of course.

I'll see if I can get some changes made to the user manual for clarification purposes.
